instant-xml-macros/src/lib.rs

```diff
@@ -1,185 +1,123 @@
 extern crate proc_macro;
 
+mod de;
+mod se;
+
+use std::collections::{BTreeSet, HashMap};
+
 use proc_macro2::TokenStream;
 use quote::quote;
-use std::collections::{BTreeSet, HashMap};
-use syn::{parse_macro_input, Lit, Meta, NestedMeta};
+use syn::parse_macro_input;
+use syn::{Lit, Meta, NestedMeta};
+
+use crate::se::Serializer;
 
 const XML: &str = "xml";
 
-enum FieldAttribute {
+pub(crate) enum FieldAttribute {
     Namespace(String),
     PrefixIdentifier(String),
 }
 
-struct Serializer {
-    default_namespace: Option<String>,
-    other_namespaces: HashMap<String, String>,
-}
-
-impl<'a> Serializer {
-    pub fn new(attributes: &'a Vec<syn::Attribute>) -> Serializer {
-        let mut default_namespace = None;
-        let mut other_namespaces = HashMap::default();
-
-        if let Some(list) = Self::retrieve_namespace_list(attributes) {
-            match list.path.get_ident() {
-                Some(ident) if ident == "namespace" => {
-                    let mut iter = list.nested.iter();
-                    if let Some(NestedMeta::Lit(Lit::Str(v))) = iter.next() {
-                        default_namespace = Some(v.value());
-                    }
+pub(crate) fn get_namespaces(
+    attributes: &Vec<syn::Attribute>,
+) -> (Option<String>, HashMap<String, String>) {
+    let mut default_namespace = None;
+    let mut other_namespaces = HashMap::default();
+
+    if let Some(list) = retrieve_attr_list("namespace", attributes) {
```

Comment:
Should do early returns and early continues here to reduce rightward drift.

Reply:
done